src/core/protocol-authorization.ts

```diff
@@ -84,10 +102,31 @@ export class ProtocolAuthorization {
    * Constructs a chain of ancestor messages
    * @returns the ancestor chain of messages where the first element is the root of the chain; returns empty array if no parent is specified.
    */
-  private static async constructAncestorMessageChain(tenant: string, recordsWrite: RecordsWrite, messageStore: MessageStore)
+  private static async constructAncestorMessageChain(
```

Comment:
Have been thinking, it's a bit odd to call the message chain "ancestor" chain in the context of a query. What if we rename to:

```suggestion
  private static async constructExistingMessageChain(
```

Reply:
I like "ancestor" since I think of the records in protocol space as tree-like, where we are either writing to or reading from a node and we want to trace back up a list of parents. "Existing" message chain doesn't invoke the same idea of traversing upward to me.